server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -402,6 +438,7 @@ private boolean isRefreshAfterCommitSafe() {
     void uploadMetadata(Collection<String> localSegmentsPostRefresh, SegmentInfos segmentInfos, ReplicationCheckpoint replicationCheckpoint)
         throws IOException {
         final long maxSeqNo = ((InternalEngine) indexShard.getEngine()).currentOngoingRefreshCheckpoint();
```

Comment:
this should be now obtained right before calling the `try (GatedCloseable<SegmentInfos> segmentInfosGatedCloseable = indexShard.getSegmentInfosSnapshot()) {`